tuning/utils/preprocessing_utils.py

```diff
@@ -0,0 +1,342 @@
+# limitations under the License.
+# Standard
+from typing import Any, Callable, Dict, Optional
+import json
+
+# Third Party
+from datasets import Dataset
+from transformers import AutoTokenizer, DataCollatorForSeq2Seq
+from trl import DataCollatorForCompletionOnlyLM
+import datasets
+
+# Local
+from tuning.config import configs
+
+
+def validate_data_args(
+    dataset_text_field: Optional[str],
+    response_template: Optional[str],
+):
+    # Dataset containing single sequence needs a single sequence and a response template
+    if dataset_text_field is None and response_template is not None:
+        raise ValueError(
+            "Needs a corresponding dataset_text_feld \
+                in which to look for response_template"
+        )
+    if response_template is None and dataset_text_field is not None:
+        raise ValueError(
+            "Since dataset_text_field is provided, \
+                needs a corresponding response template for masking"
+        )
+    # Dataset containing JSON with fields and a formatter template
+    # TO DO load JSON and check input/output field is present
+
+    # in future : pretokenized Dataset may be added.
+
+
+def get_data_trainer_kwargs(
+    training_data_path: str,
+    validation_data_path: str,
+    packing: bool,
+    response_template: Optional[str],
+    max_sequence_length: int,
+    tokenizer: AutoTokenizer,
+    dataset_text_field: Optional[str],
+) -> Dict[str, Any]:
+    """Get trainer args related to data / processing. At the moment, this consists of:
+        - the training dataset
+        - the evaluation dataset
+        - the data collator
+        - Maybe a formatting a function [only for a special case for validation]
+    The result can be kwarg expanded into the trainer initialization.
+
+    Args:
+        training_data_path: str
+            Path to the training data.
+        validation_data_path: str
+            Path to the validation data.
+        packing: bool
+            Whether or not we should apply packing or not.
+        response_template: Optional[str]
+            Response template to be used for formatting by TRL.
+        max_sequence_length: int
+            Max sequence length to be used for sequence tokenization.
+        tokenizer: AutoTokenizer
+            Loaded tokenizer object to be used by the collator.
+        dataset_text_field: Optional[str]
+            Dataset text field fto be used for formatting by TRL.
+
+    Returns:
+        Dict[str, Any]
+            Data related kwargs to be used by the SFT Trainer.
+    """
+    data_collator = get_data_collator(
+        packing, dataset_text_field, response_template, max_sequence_length, tokenizer
+    )
+    eval_dataset = None
+    data_kwargs = {}
+    if isinstance(data_collator, DataCollatorForSeq2Seq):
+        # HACK: This function is never called, but is needed to sidestep TRL's internal validation.
+        data_kwargs["formatting_func"] = lambda x: x
```

Comment:
Just a note, this shouldn't be needed anymore now that TRL has been upgraded in a separate PR and includes the change I had made, but it also doesn't hurt anything technically adds backwards compatibility, so I think for now we can leave it

Reply:
ya I think I will leave it till I finish rest of stuff and finish quality tests etc. Will keep it in mind for later